colibre/scripts/cddf.py

```diff
@@ -0,0 +1,130 @@
+import swiftsimio as sw
+import numpy as np
+import matplotlib.pyplot as plt
+from swiftsimio.visualisation.projection import project_gas
+from velociraptor.observations import load_observation
+
+from unyt import Mpc, g, cm, mp
+
+from astropy.cosmology import z_at_value
+import astropy.units as u
+
+import glob
+
+
+def plot_cddf(snapshot_filenames, names, output_path, observational_data):
+    simulation_lines = []
+    simulation_labels = []
+
+    fig, ax = plt.subplots()
+
+    for snapshot_filename, name in zip(snapshot_filenames, names):
+
+        # load the data
+        data = sw.load(snapshot_filename)
+        z = data.metadata.redshift
+        boxsize = data.metadata.boxsize[0]
+
+        # compute an appropriate number of pixels to use
+        num_pix = int(4000.0 * (boxsize / (12.5 * Mpc)))
+
+        # create the HI mass fraction dataset
+        data.gas.HI_mass_fraction = (
+            data.gas.masses
+            * data.gas.element_mass_fractions.hydrogen
+            * data.gas.species_fractions.HI
+        )
+        HI_map = project_gas(
+            data,
+            resolution=num_pix,
+            project="HI_mass_fraction",
+            parallel=True,
```

Comment:
This might be a surprise for someone who expects the pipeline to be run on a single thread.

Maybe set to `False` by default but pass a `bool` parameter to the script in the same way as in

https://github.com/SWIFTSIM/pipeline-configs/blob/master/colibre/config.yml#L221

so that the user can specify in the config file whether he/she wants to run this script in parallel?

@bwvdnbro what do you would be best?

Reply:
I actually don't know if doing this in parallel makes a huge difference. This was part of the original script. So I have no problems with not doing this in parallel. On the other hand, I can't think of many scenarios where you would want to make sure the pipeline is run on a single thread. Having more threads than idle cores is not really a problem, then it will just be less parallel than expected.

Having to change a parameter in the config file is about as easy as having to change this value in the script itself, so I'm not a huge fan of that.

Reply:
Turns out that the speed difference is quite significant, even for a 6 Mpc box.

So what if we go with your suggestion, but make the default `parallel=True`?

Reply:
I think this sounds good!